### ISSUE TITLE
CookieJar::cookiesToMap can produce cookies with incorrect expiration dates

### ISSUE THREAD

**Issue by tmallery** [ariya/phantomjs#13638]
I don't see this listed as a bug currently, sorry if this is a duplicate.

As of PhantomJS 2.0 CookieJar::cookiesToMap has a bug where any optional fields that were present in the previous cookie but not in the current cookie will get set as if the current cookie has them.

The loop iterates over cookies and stores values from each cookie in a *temporary* QVariantMap. It doesn't clear out the contents of the map between cookies though which can lead to session cookies getting dumped out with expiration dates as long as the at least one cookie before it has an expiration date.

Simple fix is to add "cookie.clear();" before it adds values to the map.

**Comment by zackw** [comment]
This sounds like a real bug in our code; congratulations!  (The overwhelming majority of bugs reported in this tracker are actually problems with Webkit.)

Would you be able to work up a self-contained test case for this bug?  Then we can be sure we are all talking about the same thing.

**Comment by tmallery** [comment]
I've created a test which will show the problem as well as verify the fix works. One thing to note is that this bug/test is dependent on the order of the cookies in CookieJar. If for example cookies were sorted by expiration and the ones without expiration dates were first then this bug wouldn't happen.
Here is a test case that will show the problem [sorry this won't let me paste the file]:
var assert = require('../../assert');

var cookies = [{
					'name':     'beforeExpires',   		/* required property */
					'value':    'expireValue',  			/* required property */
					'domain':   '.abc.com',     		/* required property */
					'path':     '/',
					'httponly': false,
					'secure':   false,
					'expires':  'Tue, 10 Jun 2025 12:28:29 GMT'
				},
				{
					'name':     'noExpiresDate',   			/* required property */
					'value':    'value',  				/* required property */
					'domain':   '.abc.com',     		/* required property */
					'path':     '/',
					'httponly': false,
					'secure':   false,
					'expires':  null
				},
				{
					'name':     'afterExpires',   		/* required property */
					'value':    'value',  				/* required property */
					'domain':   '.abc.com',     		/* required property */
					'path':     '/',
					'httponly': false,
					'secure':   false,
					'expires':  'Mon, 10 Jun 2024 12:28:29 GMT'
				}];

for(var cookieIdx in cookies)
	phantom.addCookie(cookies[cookieIdx]);

var phantomCookies = phantom.cookies;

for(var cookieIdx in phantomCookies)
{
	if( phantomCookies[cookieIdx].name === "noExpiresDate" )
	{
		var expiresDate = phantomCookies[cookieIdx].expires;
		console.log('expires date ' + expiresDate);
		assert.isTrue(expiresDate === undefined)
	}
}

If you need anything else please let me know.

**Comment by zackw** [comment]
Thanks for the nice clear test case.  We don't need anything more from you, and we'll let you know when a fix is available.  (We have very limited development manpower right now, it might be a while.  Sorry about that.  If you feel like fixing it yourself and sending in a pull request, that would probably speed things up.)

**Comment by tmallery** [comment]
I'd be happy to fix this, but I'm not sure get my change pushed such that it can be reviewed before being submitted. I can't create a pull request without having a branch to submit to. I followed the steps on https://github.com/ariya/phantomjs/blob/2.0/CONTRIBUTING.md, to create a local branch, but now I need to push the branch some how for you guys to see it.
I did this: git push -u https://github.com/ariya/phantomjs.git incorrectExpirationDates and didn't work due to permissions. I either need permission or I'm doing it wrong.
